site-src/geps/gep-1364.md

```diff
@@ -0,0 +1,328 @@
+  Gateway's gatewayClass field, or the `parentRefs` field in Route resources)
+  * Any specified TLS secrets exist
+* The resource is supported by the controller by ensuring things like:
+  * Any Kinds being referred to by the resource are supported
+  * Features being used by the resource are supported
+
+All of these rules can be summarized into:
+* The resource will produce some configuration in the underlying data plane.
+
+For Gateway, `Attached` also subsumes the functions of `Scheduled`: `Attached`
+set to `true` means that sufficient capacity exists on underlying infrastructure
+for the Gateway to be provisioned. If that capacity does not exist, then the
+Gateway cannot be reconciled successfully, and so fails to attach to the
+owning GatewayClass.
+
+Note that some classes of inter-resource reference failure do _not_ cause a resource
+to become unattached (that is, to have the `Attached` condition set to `status: false`).
+* Nonexistent Service backends - if the backend does not exist on a HTTPRoute that
+is otherwise okay, then the data plane must generate 500s for traffic that matches
+that HTTPRoute.
+* HTTPRoutes with *all* backends in other namespaces, but not permitted by ReferenceGrants.
+In this case, the "nonexistent service backends" rules apply, and 500s must be
+generated.
+
+For ReferenceGrant or not-designed-yet Policy resources, `Attached` means that:
+* the resource has a correctly-defined set of resources that it applies to
+* the resource has a syntactically and semantically valid `spec`
+
+Note that having a correctly-defined set of resources that is empty does not make
+these resources unattached.
+
+Note that for other Route types that don't have a clear mechanism like HTTP does
+for indicating a server failure (like the HTTP code 500 does), not having existing
+backends may not produce any configuration in the data plane, and so may cause
+the resource to fail to attach. (An example here could be a TCP Route with
+no backends, we need to decide if that means that a port should be opened that
+actively closes connections, or if no port should be opened.)
+
+
+#### Ready
+
+Currently, the `Ready` condition text for Gateway says:
+```go
+	// This condition is true when the Gateway is expected to be able
+	// to serve traffic. Note that this does not indicate that the
+	// Gateway configuration is current or even complete (e.g. the
+	// controller may still not have reconciled the latest version,
+	// or some parts of the configuration could be missing).
+```
+
+This is pretty unclear - how can the Gateway serve traffic if config is missing?
+In the past, we've been asked to have a Condition that only flips to `true` when
+*all* required configuration is present.
+
+For many implementations (certainly for Envoy-based ones), getting this information
+correctly and avoiding races on applying it is surprisingly difficult. For this
+reason, this GEP proposes we give the `Ready` condition a short window
```

Comment:
I think I'd rather reserve `Ready` for "Ready at this moment" and instead use a different term to represent what we're actually describing - this config has been written to the data plane but we're not sure how long that will take to propagate. Previously `Reconciled` has been proposed to communicate this.

Reply:
Okay, I'll update this section to:
- add `Ready` as an Extended support condition, that, if present, indicates that the data plane is fully configured. But because it's not always present, we'll need `Reconciled` to indicate what we have here - that everything is okay and the data plane will be ready soon.